parser-typechecker/src/Unison/Codebase/Branch.hs

```diff
@@ -501,8 +510,16 @@ empty0 =
   Branch0 mempty mempty mempty mempty mempty mempty mempty mempty mempty mempty
 
 -- | Checks whether a Branch0 is empty.
+-- This is not as easy as checking whether the branch is equal to the `empty0` branch
```

Comment:
Does "empty" mean "has no types, no terms, no patches, and no non-empty children"?

Reply:
That's the definition this currently uses yes, whereas, unintuitively, `isEmpty` requires that the branch history is also empty. Names get a bit tricky here.

Reply:
Ah, ok. Perhaps that could be documented alongside the main line, "Checks whether a Branch0 is empty".